packages/ketcher-core/src/application/render/renderers/PolymerBondRenderer.ts

```diff
@@ -36,14 +37,38 @@ export class PolymerBondRenderer extends BaseRenderer {
   }
 
   private isSnakeBondAvailableForMonomer(monomer?: BaseMonomer) {
-    return monomer instanceof Peptide || monomer instanceof Chem;
+    return (
+      monomer instanceof Peptide ||
+      monomer instanceof Chem ||
+      monomer instanceof Sugar ||
+      monomer instanceof Phosphate
+    );
+  }
+
+  private isMonomersTypeDifferent() {
+    return (
+      (this.polymerBond.secondMonomer &&
+        [Peptide, Chem].some(
+          (type) => this.polymerBond.firstMonomer instanceof type,
+        ) &&
+        [Sugar, Phosphate].some(
+          (type) => this.polymerBond.secondMonomer instanceof type,
+        )) ||
+      ([Peptide, Chem].some(
+        (type) => this.polymerBond.secondMonomer instanceof type,
+      ) &&
+        [Sugar, Phosphate].some(
+          (type) => this.polymerBond.firstMonomer instanceof type,
+        ))
+    );
```

Comment:
I would propose to move part of this logic into concrete monomer classes (Like Peptide, Chem, e t c). Like so for Peptide:

public isMonomerTypeDifferentForChaining(monomerToChain: BaseMonomer) {
  return monomerToChain instanceof Sugar || monomerToChain instanceof Phosphate
}